lib/routes/password.js

```diff
@@ -654,77 +632,44 @@ module.exports = function (
           request.emitMetricsEvent('password.forgot.verify_code.start'),
           customs.check(request, passwordForgotToken.email, 'passwordForgotVerifyCode')
         ])
-          .then(
-            function () {
-              if (butil.buffersAreEqual(passwordForgotToken.passCode, code) &&
-                  passwordForgotToken.ttl() > 0) {
-                db.forgotPasswordVerified(passwordForgotToken)
-                  .then(
-                    function (accountResetToken) {
-                      return db.accountEmails(passwordForgotToken.uid)
-                        .then((emails) => {
-                          return mailer.sendPasswordResetNotification(
-                            emails,
-                            passwordForgotToken,
-                            {
-                              code: code,
-                              acceptLanguage: request.app.acceptLanguage,
-                              flowId: flowId,
-                              flowBeginTime: flowBeginTime,
-                              uid: passwordForgotToken.uid
-                            }
-                          )
-                        })
-                        .then(
-                          function () {
-                            return request.emitMetricsEvent('password.forgot.verify_code.completed')
-                          }
-                        )
-                        .then(
-                          function () {
-                            return accountResetToken
-                          }
-                        )
-                    }
-                  )
-                  .then(
-                    function (accountResetToken) {
+          .then(async () => {
+            if (butil.buffersAreEqual(passwordForgotToken.passCode, code) &&
+                passwordForgotToken.ttl() > 0) {
+              const accountResetToken = await db.forgotPasswordVerified(passwordForgotToken)
+              const emails = await db.accountEmails(passwordForgotToken.uid)
+              await mailer.sendPasswordResetNotification(
+                  emails,
+                  passwordForgotToken,
+                  {
+                    code: code,
+                    acceptLanguage: request.app.acceptLanguage,
+                    flowId: flowId,
+                    flowBeginTime: flowBeginTime,
+                    uid: passwordForgotToken.uid
+                  }
+              )
+              await request.emitMetricsEvent('password.forgot.verify_code.completed')
+              return accountResetToken
+            }
+            else {
+              await failVerifyAttempt(passwordForgotToken)
+              return error.invalidVerificationCode({
+                tries: passwordForgotToken.tries,
+                ttl: passwordForgotToken.ttl()
+              })
 
-                      reply(
-                        {
-                          accountResetToken: accountResetToken.data
```

Comment:
@deeptibaghel we need to keep this response format, above you have response as 

`return accountResetToken`, but the expected output of this route should be 

```
return { accountResetToken: accountResetToken.data }
```